java/google/registry/rde/XjcToDomainResourceConverter.java

```diff
@@ -0,0 +1,329 @@
+        default:
+          throw new IllegalArgumentException(
+              "Unsupported grace period status: " + gracePeriodStatus.getS());
+      }
+    }
+  }
+
+  /** Converts {@link XjcRdeDomain} to {@link DomainResource}. */
+  static DomainResource convertDomain(XjcRdeDomain domain) {
+    // First create history entry and autorenew billing event/poll message
+    // Autorenew billing event is required for creating AUTO_RENEW grace period
+    HistoryEntry historyEntry = createHistoryEntry(domain);
+    BillingEvent.Recurring autoRenewBillingEvent = createAutoRenewBillingEvent(domain, historyEntry);
+    createAutoRenewPollMessage(domain, historyEntry);
+    GracePeriodConverter gracePeriodConverter = new GracePeriodConverter(
+        domain, Key.create(autoRenewBillingEvent));
+    DomainResource.Builder builder = new DomainResource.Builder()
+        .setFullyQualifiedDomainName(domain.getName())
+        .setRepoId(domain.getRoid())
+        .setIdnTableName(domain.getIdnTableId())
+        .setCurrentSponsorClientId(domain.getClID())
+        .setCreationClientId(domain.getCrRr().getValue())
+        .setCreationTime(domain.getCrDate())
+        .setRegistrationExpirationTime(domain.getExDate())
+        .setLastEppUpdateTime(domain.getUpDate())
+        .setLastEppUpdateClientId(domain.getUpRr() == null ? null : domain.getUpRr().getValue())
+        .setLastTransferTime(domain.getTrDate())
+        .setStatusValues(
+            ImmutableSet.copyOf(Iterables.transform(domain.getStatuses(), STATUS_CONVERTER)))
+        .setNameservers(convertNameservers(domain.getNs()))
+        .setGracePeriods(ImmutableSet
+            .copyOf(Iterables.transform(
+                domain.getRgpStatuses(), gracePeriodConverter)))
+        .setContacts(
+            ImmutableSet.copyOf(Iterables.transform(domain.getContacts(), CONTACT_CONVERTER)))
+        .setDsData(
+            domain.getSecDNS() == null ? ImmutableSet.<DelegationSignerData>of() : ImmutableSet
+            .copyOf(Iterables.transform(domain.getSecDNS().getDsDatas(), SECDNS_CONVERTER)))
+        .setTransferData(convertDomainTransferData(domain.getTrnData()));
+    checkArgument(
+        domain.getRegistrant() != null, "Registrant is missing for domain '%s'", domain.getName());
+    builder = builder.setRegistrant(convertRegistrant(domain.getRegistrant()));
+    return builder.build();
+  }
+
+  private static BillingEvent.Recurring createAutoRenewBillingEvent(
+    XjcRdeDomain domain, HistoryEntry historyEntry) {
+    final BillingEvent.Recurring billingEvent = new BillingEvent.Recurring.Builder()
+        .setReason(Reason.RENEW)
+        .setFlags(ImmutableSet.of(Flag.AUTO_RENEW))
+        .setTargetId(domain.getRoid())
+        .setClientId(domain.getClID())
+        .setEventTime(domain.getExDate())
+        .setRecurrenceEndTime(END_OF_TIME)
+        .setParent(historyEntry)
+        .build();
+    ofy().transact(new VoidWork() {
+
```

Comment:
Weird newline.  You could either delete it, or move it up a line to before ofy().transact(...